web/modules/custom/lit_cover_service/src/Service/BatchService.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+namespace Drupal\lit_cover_service\Service;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\file\Entity\File;
+use Drupal\node\Entity\Node;
+
+/**
+ * Class BatchService.
+ */
+class BatchService {
+
+  /**
+   * Batch delete process callback.
+   *
+   * @param int $batchId
+   *   Id of the batch
+   * @param $batchTotal
+   *   Total number of batch operations
+   * @param array $nids
+   *   The nids to process
+   * @param object $context
+   *   Context for operations
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function deleteBookCovers($batchId, $batchTotal, $nids, &$context) {
+    if (!isset($context['sandbox']['progress'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['current_node'] = 0;
+      $context['sandbox']['max'] = $batchTotal;
+    }
+
+    $context['message'] = t('Deleting batch @batchId of @batchTotal', [
+      '@batchId' => $batchId,
+      '@batchTotal' => $batchTotal,
+    ]);
+
+    $nodes = Node::loadMultiple($nids);
+    $deletedImageFiles = [];
+    foreach ($nodes as $node) {
+      $deletedImageFiles[] = self::clearImageField($node);
+      $node->save();
+
+      // Store some result for post-processing in the finished callback.
+      $context['results'][] = $node->nid;
+
+      // Update our progress information.
+      $context['sandbox']['progress']++;
+      $context['sandbox']['current_node'] = $node->nid;
+    }
+
+    self::deleteFileEntities($deletedImageFiles);
```

Comment:
`Self` vs `this`? `deleteFileEntities` is not a static function? This is true for all calls to that function in this file?

Reply:
`Self` is normally used for static object function calls.

Reply:
Also true for the other `self::` in this object.

Reply:
Fixed. Needs to be `static` for use via Drush.